feat(blog): release 2.5 blogpost (#1524)
kumahq/kuma-website: app/_posts/2023-11-14-kuma-2-5-0.md

---
title: Kuma 2.5 release with advanced locality-aware load-balancing, new policies GA and auto-reachable services
description: Kuma 2.5 release with advanced locality-aware load-balancing, new policies GA and auto-reachable services
date: 2023-11-13

> headerImage: /assets/images/blog/test-header.jpg

Author: No we use this one cause we don't have other headers

## Upgrading

We strongly suggest upgrading to Kuma 2.5.0. Upgrading is easy through `kumactl` or Helm.
Be sure to carefully read the [Upgrade Guide](https://github.com/kumahq/kuma/blob/master/UPGRADE.md) before upgrading Kuma.

## Notable features:

* 🚀 Advanced locality-aware load-balancing inside and across zones to help you achieves cost-savings and high-reliability even in the most constrained environments. 
* 🚀 Reachable services can now be derived from `MeshTrafficPermission`s to get performance improvements for free.
* 🚀 Support for Gateway API v1 following Gateway APIs first GA release! 
* 🚀 Delta KDS is now enabled by default. This greatly reduces the resource consumption of the Global CP / Zone CP protocol. 
* 🚀 Many improvements to the GUI.
* 🚀 Upgrade to Envoy 1.28.

And a lot more! Check out the full [release notes](https://github.com/kumahq/kuma/releases/tag/2.5.0) to see everything in this release.

## Advanced locality aware load balancing

As teased in a [Kong blog post](https://konghq.com/blog/product-releases/kuma-2-5) advanced locality-aware is the biggest feature of 2.5.0.

It has two main components:

> - Locality-aware load-balancing within a zone which can be helpful to save costs for zones that may spread over multiple network boundaries (AWS Availability zones for example).

Reviewer: nitpick:

```suggestion
- Locality-aware load balancing within a zone can be helpful in saving costs for zones that may span multiple network boundaries, such as AWS Availability Zones.
```

- Locality-aware load-balancing across zones which can help you achieve strong reliability while respecting some organizational constraints.

This feature is built in the existing [`MeshLoadBalancingStrategy` policy](https://kuma.io/docs/2.5.x/policies/meshloadbalancingstrategy/) and replaces the existing locality awareness that was set at the mesh level.

> Check to blogpost above, the [MADR](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/031-locality-aware-load-balancing.md) or the docs (!!! TODO docs) for more details on this new feature.

Reviewer: nitpick:

```suggestion
Check the blogpost above, the [MADR](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/031-locality-aware-load-balancing.md) or the docs (!!! TODO docs) for more details on this new feature.
```

> ## inferring reachable services from `MeshTrafficPermission`

Reviewer: nitpick: I think it's title case

```suggestion
## Inferring reachable services from `MeshTrafficPermission`
```

Service Meshes by default enable a fully connected graph of services.
This is problematic as we can't naively optimize which instance of the mesh needs to know of which services.

> The result of this is growing size of configuration and sidecar footprint becoming important.

Reviewer: nitpick:

```suggestion
The result of this is a growing size of configuration, and the sidecar footprint becoming non-negligible.
```
